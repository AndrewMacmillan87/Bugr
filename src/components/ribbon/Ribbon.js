import React from "react";

import "./ribbon.css";

export const Ribbon = () => {
    return (
        <header className='app-ribbon'>
            <div className='app-ribbon-btn-container'>
                <button className='ribbon-btn' id='file-ribbon-btn' type='button'>
                    File
                </button>
                <button className='ribbon-btn' id='edit-ribbon-btn' type='button'>
                    Edit
                </button>
            </div>
            <div className='app-visibility-controls-container'>
                <button className='app-control-btn' id='minimise' type='button'>
                    <i className='far fa-window-minimize'></i>
                </button>
                <button className='app-control-btn' id='maximise' type='button'>
                    <i className='fas fa-window-maximize'></i>
                </button>
                <button className='app-control-btn' id='quit-app' type='button'>
                    <i className='far fa-window-close'></i>
                </button>
            </div>
        </header>
    );
};

export default Ribbon;
