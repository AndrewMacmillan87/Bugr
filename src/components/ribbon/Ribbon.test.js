import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Ribbon from "./Ribbon";

describe("Ribbon component", () => {
    test("Renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Ribbon />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("Renders correctly", () => {
        const tree = renderer.create(<Ribbon />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe("Ribbon HTML elements", () => {
        it("should render a header element", () => {
            const wrapper = mount(<Ribbon />);
            expect(wrapper.find("header").length).toEqual(1);
        });

        it("should render 2 div elements", () => {
            const wrapper = mount(<Ribbon />);
            expect(wrapper.find("div").length).toEqual(2);
        });
        it("should render 5 button elements", () => {
            const wrapper = mount(<Ribbon />);
            expect(wrapper.find("button").length).toEqual(5);
        });
        it("should render 3 i elements", () => {
            const wrapper = mount(<Ribbon />);
            expect(wrapper.find("i").length).toEqual(3);
        });
    });
});
