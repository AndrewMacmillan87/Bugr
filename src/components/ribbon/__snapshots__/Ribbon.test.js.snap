// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ribbon component Renders correctly 1`] = `
<header
  className="app-ribbon"
>
  <div
    className="app-ribbon-btn-container"
  >
    <button
      className="ribbon-btn"
      id="file-ribbon-btn"
      type="button"
    >
      File
    </button>
    <button
      className="ribbon-btn"
      id="edit-ribbon-btn"
      type="button"
    >
      Edit
    </button>
  </div>
  <div
    className="app-visibility-controls-container"
  >
    <button
      className="app-control-btn"
      id="minimise"
      type="button"
    >
      <i
        className="far fa-window-minimize"
      />
    </button>
    <button
      className="app-control-btn"
      id="maximise"
      type="button"
    >
      <i
        className="fas fa-window-maximize"
      />
    </button>
    <button
      className="app-control-btn"
      id="quit-app"
      type="button"
    >
      <i
        className="far fa-window-close"
      />
    </button>
  </div>
</header>
`;
